Makefile.rules

```diff
@@ -516,6 +513,7 @@ build/main.c: src/main.c
 # build rule for the gap executable used by the `install-bin` target
 build/gap-install: libgap$(SHLIB_EXT) cnf/GAP-LDFLAGS cnf/GAP-LIBS cnf/GAP-OBJS build/obj/build/main.c.o
 	$(QUIET_LINK)$(LINK) $(GAP_LDFLAGS) build/obj/build/main.c.o $(GAP_LIBS) -L${abs_builddir} -lgap -o $@
+	@install_name_tool -change $(LIBGAP_FULL) $(libdir)/$(LIBGAP_FULL) $@
```

Comment:
This needs to be changed so it only runs on macOS

Reply:
I'll add that relying on `install_name_tool -change` is brittle unless you use the `ld` option `-headerpad_max_install_names` at the time of linking. It's typically better to set the correct install name already in the linker command.

Reply:
Thank you, you are of course 100% right.

However, setting it in the linker command would require linking two copies of the library: the one we do now and then a second copy that gets installed. That wouldn't be completely unreasonable (we already build a second `gap` executable for installation), but it would really only be necessary for macOS, overall it was just easier to hack it together this way. It seems to work reasonably well.

Thanks also for reminding me about `-headerpad_max_install_names`, I had forgotten about that, will add it later/tomorrow to this PR.